### DISCUSSION TITLE
User meetings FYI

### DISCUSSION THREAD

**Discussion by mihlema** [spine-tools/SpineOpt.jl#849] [Support]
### :loudspeaker:UPDATE: Community :rocket: 

To connect with other users, to identify problems and exchange experiences, we will organize monthly early-user meetings for Spine Toolbox and SpineOpt. The meetings will be held first Tuesday of every month at 3pm CEST [:calendar: ics zip](https://github.com/spine-tools/SpineOpt.jl/files/10497817/Spine.Toolbox.and.SpineOpt_.Exchange_.Q.A_.Help.zip)
and can be joined [here](https://teams.microsoft.com/l/meetup-join/19%3ameeting_MTA4MTZmMjMtNzk0ZS00ZjFkLWFjZjEtODVhNDc3Yjg5MjBj%40thread.v2/0?context=%7b%22Tid%22%3a%22021f8f00-6328-4916-b79c-b49b9a19a7d6%22%2c%22Oid%22%3a%22f45e2eeb-78d8-4230-903d-49e42a141be3%22%7d)..

Might be interesting for you @OliverLinsel @CyrilChambe @Mathias157 @LeoniePlaga @rputris @fbozzolo @RowanPeng. Feel free to spread the word.

**Comment by OliverLinsel** [comment]
Thank you very much Maren! Looking forward to it :)

**Comment by Mathias157** [comment]
Thanks a lot for the notification, @mihlema. I will be there!

**Comment by Mathias157** [comment]
Hi all,

Here's a DataInput sample of my renewable fuel model, that I will present in tomorrow's Early developer meeting.
Note that this probably only works with older versions of SpineOpt and SpineToolbox:
- SpineOpt v0.6.3
- SpineToolbox v0.6.5

The resolution of the model is set to 4 timesteps, for quick solving, and 6 active nodes out of 11 (spatial resolution).

[REfuelModel.zip](https://github.com/Spine-project/SpineOpt.jl/files/10115110/REfuelModel.zip)

**Comment by mihlema** [comment]
Dear all,
as of January, 10 am CET is not feasible for me anymore. I'd alternatively propose to have the meeting at 4 pm CET. Could you quickly signal with a thumbs up/down if this would work for you? (Otherwise, I'll just create a doodle)

**Comment by DillonJ** [comment]
### Parameter Defaults

A quick thought I had from last meeting... let me know if there is a better place to post such thoughts/discussions...

Use of parameter value defaults in SpineOpt should be used with a healthy degree of caution since many constraint equations are written based on the absence, presence or value of the default. It should be viewed as a developer setting. When a default  is specified, it means that every entity of the relevant class will have a value for that parameter, whether it's one explicitly set, or the default value. Some constraints filter variable and/or constraint indices by whether they have values for a certain parameters, so specifying a default for a parameter that didn't have one originally can potentially lead to decreasing performance and/or unintended behaviour. I think @mihlema experienced this before and it can be difficult to debug!

**Comment by mihlema** [comment]
> ### Parameter Defaults
> A quick thought I had from last meeting... let me know if there is a better place to post such thoughts/discussions...
> 
> Use of parameter value defaults in SpineOpt should be used with a healthy degree of caution since many constraint equations are written based on the absence, presence or value of the default. It should be viewed as a developer setting. When a default is specified, it means that every entity of the relevant class will have a value for that parameter, whether it's one explicitly set, or the default value. Some constraints filter variable and/or constraint indices by whether they have values for a certain parameters, so specifying a default for a parameter that didn't have one originally can potentially lead to decreasing performance and/or unintended behaviour. I think @mihlema experienced this before and it can be difficult to debug!

I think it's great to keep notes on these recommendations. I think we should probably include such things in the Documentation, and add a section there "best practice" or something

**Comment by mihlema** [comment]
Regarding scheduling a new regular meeting as of January, it's probably afterall easiest if we just use a [doodle](https://doodle.com/meeting/participate/id/avlyxynd).

**Comment by mihlema** [comment]
Happy New Year!
According to the doodle, a possible date for our recurring calls could be Tuesday morning at 3pm CE(S)T. If you still want to vote, go ahead. I will otherwise update the calendar invite in the coming days.

**Comment by DillonJ** [comment]
Hi folks... it looks the calendar invite wasn't updated. Tuesday 3pm CE(S)T works for me... I propose updating the calendar invite

**Comment by DillonJ** [comment]
Hi folks - I edited the original post with an updated calendar invite for  Tuesday 3pm CE(S)T

**Comment by mihlema** [comment]
Thanks for taking care of this @DillonJ !

**Comment by mihlema** [comment]
@lilanger sorry for the confusion. This is the issue where we update on changes. Also, JFYI, linking you to the issue too @RowanPeng, @josefkoell,

**Comment by mihlema** [comment]
Tomorrow is going to be our next meeting! looking forward to talking to you all. Also in case you missed it, spine-tools is now on [gitter](https://matrix.to/#/#spine-tools_community:gitter.im) :)

**Comment by mihlema** [comment]
In case you haven't seen these before, there are two recent video tutorials on Spine Toolbox and SpineOpt available:
[SpineOpt](https://www.youtube.com/watch?v=brf6rlzNDR0)
[Spine Toolbox](https://www.youtube.com/watch?v=oRwXwV9tuSo)

**Comment by lilanger** [comment]
Hi all
I also created the two repositories we talked about last time, for standardized results and visualization and cluster runs:
https://github.com/spine-tools/spine-DTU-visualization
https://github.com/spine-tools/spine-cluster

Hope you find them helpful.

**Comment by mihlema** [comment]
Hi all,
not sure if anyone is trying to join the call today. @DillonJ in case you're around, the meeting can't be started without you I am afraid. I am guessing that probably many people are on holidays anyways this week, so I'd suggest to just skip this instance otherwise?

**Comment by RowanPeng** [comment]
Hi,

Thanks for this email. I would like to join the meeting, if you still
schedule one.

Regards,

Peng

Maren Ihlemann ***@***.***> 于2023年8月1日周二 14:07写道：

**Comment by mihlema** [comment]
Hi all,
just coming of our monthly call, and wanting to make sure that those calls are still useful for everyone and would like to hear your feedback on those calls, and how we can improve them going forward. Feel free to post here, or if you prefer, you can also reach out to me via gitter (@mihlema-5c0540c5d73408ce4fb0cdae:gitter.im) or e-mail.

**Comment by lilanger** [comment]
Hi all,
I enjoy these calls quite a lot and find them super helpful. To get updates from the Mopo project is very interesting, that would be nice to have on a regular basis. I guess the amount of questions comes in waves (also with the academic year). It would be nice to see some of the work people have been doing using SpineOpt, so if people would come back to share their success stories in a way.

**Comment by mihlema** [comment]
Hi all,
as discussed during our last meeting, we will reconvene next week Tuesday for our next user meeting. @DillonJ we should just be able to use the "standard" link, I guess? Otherwise, could you possibly send another invite here for the updated meeting? Thanks!

**Comment by mihlema** [comment]
https://teams.microsoft.com/l/meetup-join/19%3ameeting_MTA4MTZmMjMtNzk0ZS00ZjFkLWFjZjEtODVhNDc3Yjg5MjBj%40thread.v2/0?context=%7b%22Tid%22%3a%22021f8f00-6328-4916-b79c-b49b9a19a7d6%22%2c%22Oid%22%3a%22f45e2eeb-78d8-4230-903d-49e42a141be3%22%7d

**Comment by lilanger** [comment]
Just adding the colleagues from CBS for updates: @weibezahn @hannesfelipe

**Comment by mihlema** [comment]
Hi all,
thanks for joining our call today! Quick summary of action items:
- Next meeting, we are planning to have topical meeting on **SpineOpt & Hydrogen**
    - The idea would be that everyone who's currently working on hydrogen/Spine is bringing a quick overview of their respective projects, scope, modelling workflows/approach, what's the key challenge/novelty your want to look at (depending on how much you want to share). What has been useful/challenging using SpineOpt etc. 
    - Ideally, people can share experiences, get inspired, find synergies etc. etc.
    - I think we could aim for 5' per project - but it can be more/less depending on how advanced/early your efforts are and how comfortable you are with presenting :)
- We are thinking about creating a **mailing list** to make it easier for folks to get updates etc. on meetings, changes, community outreach. We're in the process of figuring out the best way of doing this, stay tuned :)
- We'll have our next project meeting in September/October and are thinking of taking on an **in-person user meeting**. Location and exact date are still to be decided on, but if you're interested in joining, you can react to this comment with a 👍 to give us an idea, if there is an overall interest in this type of event :)

**Reply by hannesfelipe** [reply]
hej, 

in the next meeting, we would like to present our CSEI approach to model energy hubs focusing on hydrogen and power-to-x fuels.

**Comment by lilanger** [comment]
Hi all,
I think the calendar invite ended last month. Maybe you could extend it and post the link here? 
Unfortunately, I will not be able to join the March meeting since I am boarding a ferry to Norway.

But from DTU, 

- we earlier shared Mathias' **hydrogen/e-fuels in the Nordics** model [above](https://github.com/spine-tools/SpineOpt.jl/discussions/849#discussioncomment-8111100). The paper will be published very soon in Energy.
- we published a model/paper on the implications of **hydrogen regulation in Danish energy hubs** ([paper](https://orbit.dtu.dk/en/publications/renewable-fuel-regulation-implications-for-e-fuel-production-infr), [preprint](https://orbit.dtu.dk/files/345596135/2309.12747.pdf), [model](https://github.com/lilanger/Enerhub2X.spineopt))
- my students presented their **hydrogen/ammonia production in Greenland using hydropower** model in one of our meetings. They shared their [**PowerBI visualization tool**](https://github.com/spine-tools/SpineOpt.jl/discussions/849#discussioncomment-8111112) and their [model and presentation](https://github.com/JaKehrl/HydropowerP2X.SpineOpt/tree/main)

And finally, if you would like to present your work, you could still submit to our **EURO conference session** on hydrogen modelling and regulation. The submission deadline was extended to March 17. You only need an abstract and a presentation in the beginning of July.

"Dear All,

As already communicated to some of you, we will be organizing a session on Hydrogen in the Nordics (working title) within the Energy Markets stream at next year's [33rd European Conference on Operational Research](https://euro2024cph.dk/), hosted at the Technical University of Denmark (DTU) in Copenhagen from June 30 to July 3, 2024.

Papers should focus on applications of **Operations Research to energy and electricity markets with a focus on hydrogen and PtX fuels, e.g., in energy system models or energy hub models, and/or on hydrogen regulation/power purchase conditions (three pillars...).**

If you would like to contribute, please submit your abstract by March 17, 2024, at https://www.euro-online.org/conf/euro33/, using the **invitation code 33f4d039**.

Feel free to forward this invitation to any colleagues you think might be interested.

Thank you very much and best,
\Jens and Lissy"

Hope you have a great meeting and I will see some at the conference,
Lissy

**Comment by mihlema** [comment]
﻿Hi all, 

Thanks for presenting Johannes, looking forward to hearing more about your project. Thanks also Lissy for referencing your work here. @DillonJ, can you extend the meeting series and re-share the invite link here? 

Thanks!See you tomorrow :)

**Comment by mihlema** [comment]
Hi all, @DillonJ can you extend the meeting series? 
Alternatively, I'm not sure if maybe someone else can share a one-time meeting link we can all hop on.

**Comment by mihlema** [comment]
Quick update, this meeting link works:
https://teams.microsoft.com/l/meetup-join/19%3ameeting_MTA4MTZmMjMtNzk0ZS00ZjFkLWFjZjEtODVhNDc3Yjg5MjBj%40thread.v2/0?context=%7b%22Tid%22%3a%22021f8f00-6328-4916-b79c-b49b9a19a7d6%22%2c%22Oid%22%3a%22f45e2eeb-78d8-4230-903d-49e42a141be3%22%7d

**Comment by tarskul** [comment]
I've fixed the meeting schedule/invite (you should be able to enter even if the original organiser is not present) and I've updated the link in the readme i.e. [teams link](https://teams.microsoft.com/l/meetup-join/19%3ameeting_ZDBkZjBhNmEtMGFmZi00ZDllLWE4ZmQtMWJhMWZjYjUyNDUy%40thread.v2/0?context=%7b%22Tid%22%3a%223973589b-9e40-4eb5-800e-b0b6383d1621%22%2c%22Oid%22%3a%224c379655-31a5-4c3e-b6d6-b61817a44e14%22%7d). Hopefully this fixes some of the issues we had before.

**Comment by tarskul** [comment]
According to the schedule the next one is right after easter monday. I don't think that is going to be a popular time slot. For now I'm leaving it at that. Because as I said you should be able to enter regardless of who is present (can someone check though?). But you can reply to this message if you'd rather move it outside of the easter break.

**Comment by hannesfelipe** [comment]
I will miss the meeting tomorrow. Unfortunately, we have to move our presentation of the hydrogen and electrolysis modelling approach once again.

**Reply by tarskul** [reply]
No problem, I've received message from multiple people (including myself and Maren) that they will not be able to attend.

**Comment by tarskul** [comment]
For the past years, the development of Spine tools was funded by the EU-funded project Mopo. We've been mostly working quietly on this project in the background. Now, we want to show you some of the cool things we've done. This includes improvements in Spine Toolbox and SpineOpt as well as practical case studies on European and national scale. We've created a yearlong program of webinars and in person events to (re)connect and we warmly welcome whoever wants to join the party. You can find the program on the [spine tools project page](https://github.com/spine-tools).
